traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -629,117 +860,352 @@ WHERE
 		}
 		return nil, http.StatusInternalServerError, nil, fmt.Errorf("querying delivery service ID %d: %s", *dsV13.ID, err.Error())
 	}
-	if dsV13.DeepCachingType != nil {
-		*dsV13.DeepCachingType = tc.DeepCachingTypeFromString(string(*dsV13.DeepCachingType))
+	if dsV13.DeepCachingType != nil {
+		*dsV13.DeepCachingType = tc.DeepCachingTypeFromString(string(*dsV13.DeepCachingType))
+	}
+
+	res, status, userErr, sysErr := updateV13(w, r, inf, &dsV13)
+	if res != nil {
+		return &res.DeliveryServiceNullableV12, status, userErr, sysErr
+	}
+	return nil, status, userErr, sysErr
+}
+
+func updateV13(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, reqDS *tc.DeliveryServiceNullableV13) (*tc.DeliveryServiceNullableV13, int, error, error) {
```

Comment:
This is kind of confusing to review since it seems like this was all added as new code, but I think things were just moved around.

For the update* functions, you should just rename `updateV30` to `updateV31`, and change it to take in `DeliveryServiceV31` as input and output. Then _within_ the function you should immediately cast it to `DeliveryServiceNullableV30`. At the end of the function, you cast it back to `DeliveryServiceV31` before returning it.

Then you can basically copy `updateV15` to `updateV30`, change it to take `DeliveryServiceV30` as input and output, then change it to read the new `max_request_header_bytes` from the DB before passing it to `updateV31`, where the real handling is done.

Reply:
Yeah this is not a new change, it was just moved around.